runs/atreboot.sh

```diff
@@ -433,18 +433,15 @@ at_reboot() {
 		mosquitto_pub -r -t openWB/global/awattar/pricelist -m ""
 	fi
 
-	# set upload limit in php
-	#prepare for Buster
-	echo -n "fix upload limit..."
-	if [ -d "/etc/php/7.0/" ]; then
-		echo "OS Stretch"
-		sudo /bin/su -c "echo 'upload_max_filesize = 300M' > /etc/php/7.0/apache2/conf.d/20-uploadlimit.ini"
-		sudo /bin/su -c "echo 'post_max_size = 300M' >> /etc/php/7.0/apache2/conf.d/20-uploadlimit.ini"
-	elif [ -d "/etc/php/7.3/" ]; then
-		echo "OS Buster"
-		sudo /bin/su -c "echo 'upload_max_filesize = 300M' > /etc/php/7.3/apache2/conf.d/20-uploadlimit.ini"
-		sudo /bin/su -c "echo 'post_max_size = 300M' >> /etc/php/7.3/apache2/conf.d/20-uploadlimit.ini"
-	fi
+	# Check upload Limit in ALL installed php Version. 
+	echo  "check php upload limit..."
+	for d in /etc/php/*/apache2/conf.d ; do
+		fn="$d/20-uploadlimit.ini"
+		if [ ! -f "$fn" ]; then
```

Comment:
Durch das `if` änderst du das Verhalten. Früher wurde die Datei immer überschrieben. Dadurch wurden auch vorhandene Werte ersetzt. Wenn die Datei jetzt bereits existiert (mit falschen, zu kleinen Werten), dann passiert nichts.

Reply:
Und wenn jemand heimlich auf Bullseye updates bricht alles zusammen....
Also bleib mal auf dem Boden.
Wir haben es mit einem Stetch/Buster in bekannter Konfiguration zu tun.
Klar werden nicht alle eventualitäten abgedeckt.
Kein andere würde diese Datei erzeugen und wenn wird sie mal Updaten müssen dann
löschen wir 20-uploadlinit.ini und erzeugen (einmalig) 21-uploadinit.ini

Reply:
> Also bleib mal auf dem Boden.

Tu mal nicht so als hätte ich ein riesen Fass aufgemacht. Ich weise lediglich darauf hin, dass sich das Verhalten ändert. Ich mache nur auf die Tatsache aufmerksam. Ob das Auswirkungen hat habe ich nicht gesagt und weiß ich auch nicht. Ich weiß zum Beispiel nicht ob in einer älteren Version die Datei mal mit anderen Werten existierte und damit das Update nicht mehr funktionieren würde, wenn man dummerweise von der falschen Version kommt. Wenn du das alles im Blick hast, ist ja gut.

Reply:
Ich habe gerade mal nachgeschaut und ja: Es war mal anders:

https://github.com/snaptec/openWB/blob/d0c1d9311808630982810b6d2067ba3335e87a07/openwb-install.sh#L148

https://github.com/snaptec/openWB/blob/d0c1d9311808630982810b6d2067ba3335e87a07/runs/atreboot.sh#L2917

Wenn du mit einem Update zum Beispiel von dieser Version kommst, dann funktioniert es nicht.

Reply:
Ok. dass hatt ich so nicht auf dem Schirm (lag vor meiner Zeit)
Wenn also auch die Steinzeit (>2Jahre) dabei ist und dazwischen kein Update auf die Stabel17 u.folgende  durchgefuhrt wurde, dann hat du recht.
Dann eben doch sofort ein Update auf 21 um sicheren Boden unter die Füsse zu bekommen.

Reply:
Warum so kompliziert? Warum überschreibst du die Datei nicht einfach?